conwayste/src/ui/common.rs

```diff
@@ -25,22 +25,18 @@ use crate::constants::DEFAULT_UI_FONT_SCALE;
 #[macro_export]
 macro_rules! widget_from_id {
     ($type:ident) => {
-        use super::layer::Layer;
+        use super::layer::Layering;
 
         impl $type {
-            pub fn widget_from_id(layer: &mut Layer, id: WidgetID) -> Option<&mut $type>
+            pub fn widget_from_id(layer: &mut Layering, id: WidgetID) -> Option<&mut $type>
             {
-                let layer_id = layer.id();
                 let widget_result = layer.get_widget_mut(id);
                 match widget_result {
                     Ok(widget) => {
                         return widget.downcast_mut::<$type>();
                     }
                     Err(e) => {
-                        info!("Could not find $type widget of {:?} in layer of {:?}! {:?}",
-                        id,
-                        layer_id,
-                        e);
+                        info!("Could not find $type widget of {:?} in layer! {:?}", id, e);
```

Comment:
I'm not really a fan of macros that add stuff like this (log lines, etc.). You can address here or add a TODO (since this was already there before your PR).

Reply:
Agreed it should be an error.